src/Mvc/Mvc.Api.Analyzers/src/ActualApiResponseMetadataFactory.cs

```diff
@@ -84,9 +86,23 @@ void AnalyzeResponseExpression(IReturnOperation returnOperation)
         }
 
         var defaultStatusCodeAttribute = statementReturnType
-            .GetAttributes(symbolCache.DefaultStatusCodeAttribute, inherit: true)
+            .GetAttributes(defaultStatusCodeAttributeSymbol, inherit: true)
             .FirstOrDefault();
 
+        // If the type is not annotated with a default status code, then examine
+        // the attributes on any invoked method returning the type.
+        if (defaultStatusCodeAttribute is null && returnedValue.Syntax is InvocationExpressionSyntax targetInvocation)
```

Comment:
The precedence is here a little wonky. I would think putting something on the method would be more explicit than putting it on the type. Should we prefer an attribute on the method when specified over the attribute on the type? For instance, the method could do things like configure a non-default status code.

Reply:
We already prioritize the attribute on the method when specified (e.g. the user is calling `MethodThatReturnsIActionResult` in the scenario below) over the attribute in the type returned inside `MethodThatReturnsIActionResult`.

Are you referring to being able to use the `DefaultStatusCode` attribute on the action method?

```
public IActionResult MethodThatReturnsIActionResult()
{
   return FooResult();
}

public class FooResult
{
}

public IActionResult GetFoo()
{
  return MethodThatReturnsIActionResult();
}

public IActionResult GetFoo2()
{
  return new FooResult();
}
```

Reply:
I guess I was going for this: 

```C#
[DefaultStatusCode(421)] public override BadActionResult BadResult() => new StatusCodeResult(421);
```

Looking at the code, it looked we would infer the default as 400 (because that's what BadActionResult is annotated with) rather than 421 even though it's "closer" to the callsite.